packages/formdescriptioneditors/frontend/sirius-components-formdescriptioneditors/src/SplitButtonWidget.tsx

```diff
@@ -0,0 +1,350 @@
+      }
+      if (addWidgetData) {
+        const { addWidget } = addWidgetData;
+        if (isErrorPayload(addWidget)) {
+          setState((prevState) => {
+            return { ...prevState, message: addWidget.message };
+          });
+        }
+      }
+    }
+  }, [addWidgetLoading, addWidgetData, addWidgetError]);
+
+  useEffect(() => {
+    if (widgetRef.current && selection.entries.find((entry) => entry.id === widget.id)) {
+      widgetRef.current.focus();
+      setState((prevState) => ({ ...prevState, selected: true }));
+    } else {
+      setState((prevState) => ({ ...prevState, selected: false }));
+    }
+  }, [selection, widget]);
+
+  const classes = useStyles();
+
+  const handleMenuItemClick = (_event, index) => {
+    setState((prevState) => ({ ...prevState, open: false, selectedIndex: index }));
+  };
+
+  const handleToggle = () => {
+    setState((prevState) => ({ ...prevState, open: !prevState.open }));
+  };
+
+  const handleClose = (event) => {
+    if (widgetRef.current && widgetRef.current.contains(event.target)) {
+      return;
+    }
+    setState((prevState) => ({ ...prevState, open: false }));
+  };
+
+  const handleDragEnter: React.DragEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault();
+    event.currentTarget.classList.add(classes.dragOver);
+  };
+
+  const handleDragOver: React.DragEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault();
+    event.currentTarget.classList.add(classes.dragOver);
+  };
+
+  const handleDragLeave: React.DragEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault();
+    event.currentTarget.classList.remove(classes.dragOver);
+  };
+
+  const handleDrop: React.DragEventHandler<HTMLDivElement> = (event) => {
+    event.preventDefault();
+    event.currentTarget.classList.remove(classes.dragOver);
+
+    const id: string = event.dataTransfer.getData('draggedElementId');
+    const type: string = event.dataTransfer.getData('draggedElementType');
+    if (type !== 'Widget') {
+      return;
+    }
+
+    if (id === 'Button') {
+      const addWidgetInput: GQLAddWidgetInput = {
+        id: crypto.randomUUID(),
+        editingContextId,
+        representationId,
+        containerId: widget.id,
+        kind: id,
+        index: widget.actions.length || 0,
+      };
+      const addWidgetVariables: GQLAddWidgetMutationVariables = { input: addWidgetInput };
+      addWidget({ variables: addWidgetVariables });
+    }
+  };
+
+  const getStyle = (index: number): React.CSSProperties => {
+    if (widget.actions.length == 0) {
+      const defaultProps = {
+        backgroundColor: null,
+        foregroundColor: null,
+        fontSize: null,
+        italic: null,
+        bold: null,
+        underline: null,
+        strikeThrough: null,
+      };
+      return actionStyle(theme, defaultProps);
+    }
+    return actionStyle(theme, widget.actions[index].style);
+  };
+
+  const getIconSrc = (originalSrc: string): string => {
```

Comment:
Sorry, I didn't see that before. You shouldn't do that. We don't want to try to extract something after `aql:` if the `imageURL ` of a button contains `aql:`. See how it is done in `ButtonWidget.tsx`.